[media-controls] remove the RangeControl class and all of its usage

graouts opened the pull request:
#### e55176eb756bbb7367f181c1cf48bec9fb886718
<pre>
[media-controls] remove the RangeControl class and all of its usage
<a href="https://bugs.webkit.org/show_bug.cgi?id=241976">https://bugs.webkit.org/show_bug.cgi?id=241976</a>

Reviewed by Devin Rousso.

We no longer need the RangeControl class, so let&apos;s get rid of it and its
BrightnessButton and VolumeButton subclasses. Since those subclasses are
gone, we can also remove the support objects working with it: BrightnessSupport
and MuteAndVolumeSupport. Finally, the brightness icon is no longer used
either.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/modern-media-controls/controls/brightness-button.js: Removed.
* Source/WebCore/Modules/modern-media-controls/controls/icon-service.js:
* Source/WebCore/Modules/modern-media-controls/controls/media-controls.js:
(MediaControls.):
(MediaControls.prototype.set usesLTRUserInterfaceLayoutDirection):
(MediaControls.prototype.commitProperty):
(MediaControls):
(MediaControls.prototype.get usesSingleMuteAndVolumeButton): Deleted.
* Source/WebCore/Modules/modern-media-controls/controls/range-button.css: Removed.
* Source/WebCore/Modules/modern-media-controls/controls/range-button.js: Removed.
* Source/WebCore/Modules/modern-media-controls/controls/volume-button.js: Removed.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Brightness.svg: Removed.
* Source/WebCore/Modules/modern-media-controls/images/iOS/Brightness.svg: Removed.
* Source/WebCore/Modules/modern-media-controls/images/macOS/Brightness.svg: Removed.
* Source/WebCore/Modules/modern-media-controls/js-files:
* Source/WebCore/Modules/modern-media-controls/media/brightness-support.js: Removed.
* Source/WebCore/Modules/modern-media-controls/media/media-controller.js:
(MediaController.prototype._supportingObjectClasses):
* Source/WebCore/Modules/modern-media-controls/media/mute-and-volume-support.js: Removed.
* Source/WebCore/PlatformGLib.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/251869@main">https://commits.webkit.org/251869@main</a>
</pre>

dcrousso commented on Source/WebCore/DerivedSources.make:
```diff
@@ -1768,7 +1768,6 @@ MODERN_MEDIA_CONTROLS_STYLE_SHEETS = \
     $(WebCore)/Modules/modern-media-controls/controls/media-controls.css \
     $(WebCore)/Modules/modern-media-controls/controls/media-document.css \
     $(WebCore)/Modules/modern-media-controls/controls/placard.css \
-    $(WebCore)/Modules/modern-media-controls/controls/range-button.css \
```
I think you also have to remove these from `Source/WebCore/CMakeLists.txt` and `Source/WebCore/DerivedSources-input.xcfilelist`.

graouts committed:
[media-controls] remove the RangeControl class and all of its usage
https://bugs.webkit.org/show_bug.cgi?id=241976

Reviewed by Devin Rousso.

We no longer need the RangeControl class, so let's get rid of it and its
BrightnessButton and VolumeButton subclasses. Since those subclasses are
gone, we can also remove the support objects working with it: BrightnessSupport
and MuteAndVolumeSupport. Finally, the brightness icon is no longer used
either.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/modern-media-controls/controls/brightness-button.js: Removed.
* Source/WebCore/Modules/modern-media-controls/controls/icon-service.js:
* Source/WebCore/Modules/modern-media-controls/controls/media-controls.js:
(MediaControls.):
(MediaControls.prototype.set usesLTRUserInterfaceLayoutDirection):
(MediaControls.prototype.commitProperty):
(MediaControls):
(MediaControls.prototype.get usesSingleMuteAndVolumeButton): Deleted.
* Source/WebCore/Modules/modern-media-controls/controls/range-button.css: Removed.
* Source/WebCore/Modules/modern-media-controls/controls/range-button.js: Removed.
* Source/WebCore/Modules/modern-media-controls/controls/volume-button.js: Removed.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Brightness.svg: Removed.
* Source/WebCore/Modules/modern-media-controls/images/iOS/Brightness.svg: Removed.
* Source/WebCore/Modules/modern-media-controls/images/macOS/Brightness.svg: Removed.
* Source/WebCore/Modules/modern-media-controls/js-files:
* Source/WebCore/Modules/modern-media-controls/media/brightness-support.js: Removed.
* Source/WebCore/Modules/modern-media-controls/media/media-controller.js:
(MediaController.prototype._supportingObjectClasses):
* Source/WebCore/Modules/modern-media-controls/media/mute-and-volume-support.js: Removed.
* Source/WebCore/PlatformGLib.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/251869@main
